src/datastore/postgres-store.ts

```diff
@@ -2180,8 +2180,10 @@ export class PgDataStore
     }
     const cutoffBlockHeight = cutoffResults.rows[0].block_height;
     // Delete every mempool tx that came before that block.
+    // TODO: Use DELETE instead of UPDATE once we implement a non-archival API replay mode.
     const deletedTxResults = await client.query<{ tx_id: Buffer }>(
-      `DELETE FROM mempool_txs
+      `UPDATE mempool_txs
+      SET pruned = TRUE
```

Comment:
Do we also want to set the status like here?
https://github.com/hirosystems/stacks-blockchain-api/blob/8f56087e8ff399dd6359111fa623b5a76fcc8204/src/datastore/postgres-store.ts#L3466-L3468

We could use `DroppedStaleGarbageCollect` (the one we receive from the stacks-node), although maybe a new status code for "dropped by API because stale" would be more useful for later analysis of the table?

Reply:
I think Github might have sent you an old commit email, here's the one where I set it to `DroppedStaleGarbageCollect ` @zone117x https://github.com/hirosystems/stacks-blockchain-api/pull/1101/commits/ef08c0f76184b18e3218ee207165890c9b519b33 😄

Reply:
however, your idea of a new status is great (garbage collected by the API)... I'll add that

Reply:
Haha gotcha, looks great!